specification/under_development/output_schemas/Labeled_And_Async.md

```diff
@@ -1,2 +1,238 @@
+
+## Examples
+
+### Basic Single Item Output 
+
+The QIR program would include a single call to the runtime function matching the type of the return value: 
+
+```llvm
+@0 = internal constant [3 x i8] c"0_i\00"
+call void @__quantum__rt__integer_record_output(i64 %5, i8* getelementptr inbounds ([3 x i8], [3 x i8]* @0, i32 0, i32 0))
+ret void
+```
+
+Such that the provider output for `3` shots would be: 
+
+```log
+START
+METADATA\tmetadata1_name_only
+METADATA\tmetadata2_name\tmetadata2_value
+OUTPUT\tINT\t42\t0_i
+END\t0
+START
+METADATA\tmetadata1_name_only
+METADATA\tmetadata2_name\tmetadata2_value
+OUTPUT\tINT\t41\t0_i
+END\t0
+START
+METADATA\tmetadata1_name_only
+METADATA\tmetadata2_name\tmetadata2_value
+OUTPUT\tINT\t42\t0_i 
+END\t0
+```
+
+## Measurement Result Array Output 
+
+For an array of measurement results (or classical register), the QIR program would include array start and end calls, with intervening result record calls (shown with static result allocation transformations performed): 
+
+```llvm
+@0 = internal constant [5 x i8] c"0_0a\00"
+@1 = internal constant [7 x i8] c"1_0a0r\00"
+@2 = internal constant [5 x i8] c"2_1a\00"
+@3 = internal constant [7 x i8] c"3_1a0r\00"
+@4 = internal constant [7 x i8] c"4_1a1r\00"
+call void @__quantum__rt__array_record_output(i64 1, i8* getelementptr inbounds ([5 x i8], [5 x i8]* @0, i32 0, i32 0))
+call void @__quantum__rt__result_record_output(%Result* %2, i8* getelementptr inbounds ([7 x i8], [7 x i8]* @1, i32 0, i32 0))
+call void @__quantum__rt__array_record_output(i64 2, i8* getelementptr inbounds ([5 x i8], [5 x i8]* @2, i32 0, i32 0))
+call void @__quantum__rt__result_record_output(%Result* nonnull inttoptr (i64 0 to %Result*), i8* getelementptr inbounds ([7 x i8], [7 x i8]* @3, i32 0, i32 0))
+call void @__quantum__rt__result_record_output(%Result* nonnull inttoptr (i64 0 to %Result*), i8* getelementptr inbounds ([7 x i8], [7 x i8]* @4, i32 0, i32 0))
+ret void
+```
+
+This would produce provider output for `3` shots of the form: 
+
+```log
+START
+METADATA\tmetadata1_name_only
+METADATA\tmetadata2_name\tmetadata2_value
+OUTPUT\tARRAY\t1\t0_0a
+OUTPUT\tRESULT\t0\t1_0a0r
+OUTPUT\tARRAY\t2\t2_1a
+OUTPUT\tRESULT\t1\t3_1a0r
+OUTPUT\tRESULT\t1\t4_1a1r
+END\t0
+START
+METADATA\tmetadata1_name_only
+METADATA\tmetadata2_name\tmetadata2_value
+OUTPUT\tARRAY\t1\t0_0a
+OUTPUT\tRESULT\t1\t1_0a0r
+OUTPUT\tARRAY\t2\t2_1a
+OUTPUT\tRESULT\t1\t3_1a0r
+OUTPUT\tRESULT\t1\t4_1a1r
+END\t0
+START
+METADATA\tmetadata1_name_only
+METADATA\tmetadata2_name\tmetadata2_value
+OUTPUT\tARRAY\t1\t0_0a
+OUTPUT\tRESULT\t0\t1_0a0r
+OUTPUT\tARRAY\t2\t2_1a
+OUTPUT\tRESULT\t1\t3_1a0r
+OUTPUT\tRESULT\t1\t4_1a1r
+END\t0
+```
+
+## Tuple Output 
+
+Recording tuple output works much the same way as array output, but with a different delimiter character in the final processed output. So, a QIR program that returns a tuple of a measurement result and calculated double value would end with: 
```

Comment:
Mentioning the different delimiter character in the final processed output might be getting too much into the details of the labeling schema.